azkaban-common/src/main/java/azkaban/executor/container/ContainerizedExecutionManager.java

```diff
@@ -0,0 +1,383 @@
+
+    } catch (final ExecutorManagerException e) {
+      logger.error(
+          "Failed to check if the flow is running for project " + projectId + ", flow " + flowId,
+          e);
+    }
+    return isRunning;
+  }
+
+  /**
+   * Get execution ids of all running (unfinished) flows from database.
+   */
+  public List<Integer> getRunningFlowIds() {
+    final List<Integer> allIds = new ArrayList<>();
+    try {
+      getExecutionIdsHelper(allIds, this.executorLoader.fetchUnfinishedFlows().values());
+    } catch (final ExecutorManagerException e) {
+      this.logger.error("Failed to get running flow ids.", e);
+    }
+    return allIds;
+  }
+
+  public List<Integer> getQueuedFlowIds() {
+    final List<Integer> allIds = new ArrayList<>();
+    try {
+      getExecutionIdsHelper(allIds, this.executorLoader.fetchQueuedFlows(Status.READY));
+    } catch (final ExecutorManagerException e) {
+      this.logger.error("Failed to get queued flow ids.", e);
+    }
+    return allIds;
+  }
+
+  @Override
+  public long getQueuedFlowSize() {
+    return getQueuedFlowIds().size();
+  }
+
+  @Override
+  public DispatchMethod getDispatchMethod() {
+    return DispatchMethod.CONTAINERIZED;
+  }
+
+  @Override
+  public void start() {
+    this.queueProcessor = setupQueueProcessor();
+    this.queueProcessor.start();
+  }
+
+  private QueueProcessorThread setupQueueProcessor() {
+    return new QueueProcessorThread(
+        this.azkProps, this.executorLoader);
+  }
+
+  @Override
+  public Status getStartStatus() {
+    return Status.READY;
+  }
+
+  @Override
+  public void shutdown() {
+    if (null != this.queueProcessor) {
+      this.queueProcessor.shutdown();
+    }
+  }
+
+  @Override
+  public void enableQueueProcessorThread() {
+    this.queueProcessor.setActive(true);
+  }
+
+  @Override
+  public void disableQueueProcessorThread() {
+    this.queueProcessor.setActive(false);
+  }
+
+  public State getQueueProcessorThreadState() {
+    return this.queueProcessor.getState();
+  }
+
+  public boolean isQueueProcessorThreadActive() {
+    return this.queueProcessor.isActive();
+  }
+
+  /**
+   * This QueueProcessorThread will fetch executions from database in batch and dispatch it in
+   * container.
+   */
+  private class QueueProcessorThread extends Thread {
+
+    private final long queueProcessorWaitInMS;
+    private final Props azkProps;
+    private volatile boolean shutdown = false;
+    private volatile boolean isActive = true;
+    private ExecutorService executorService;
+    private ExecutorLoader executorLoader;
+    private boolean executionsBatchProcessingEnabled;
+    private int executionsBatchSize;
+
+    public QueueProcessorThread(final Props azkProps, final ExecutorLoader executorLoader) {
+      this.azkProps = azkProps;
+      this.executorLoader = executorLoader;
+      setActive(
+          this.azkProps.getBoolean(Constants.ConfigurationKeys.QUEUEPROCESSING_ENABLED, true));
```

Comment:
Why is this config required, shouldn't this always be true ?

Reply:
There is a way to enable or disable queue processor. If you want to keep it disable and later on start it, there is a way.

Reply:
What's the use-case for disabling the queue processor once started ?

Reply:
I personally don't know. I kept the behavior same as earlier dispatch methods. I will add TODO to investigate it.